sympy/combinatorics/perm_groups.py

```diff
@@ -678,6 +680,135 @@ def basic_transversals(self):
             self.schreier_sims()
         return self._transversals
 
+    def coset_transversal(self, H):
+        """Return a transversal of the right cosets of self by its subgroup H
+        using the second method described in [1], Subsection 4.6.7
+
+        """
+
+        if not H.is_subgroup(self):
+            raise ValueError("The argument must be a subgroup")
+
+        if H.order() == 1:
+            return self._elements
+
+        self._schreier_sims(base=H.base) #so that G.base is an extension of H.base
+
+        base = self.base
+        base_ordering = _base_ordering(base, self.degree)
+        identity = Permutation(self.degree - 1)
+
+        transversals = self.basic_transversals[:]
+        # transversals is a dictionary. Get rid of the keys so that it is a list
+        # of lists and sort each list in the increasing order of base[l]^x
+        for l, t in enumerate(transversals):
+            transversals[l] = sorted(t.values(), key = lambda x: base_ordering[base[l]^x])
+
+        orbits = H.basic_orbits
+        h_stabs = H.basic_stabilizers
+        g_stabs = self.basic_stabilizers
+
+        indices = [x.order()/y.order() for x, y in zip(g_stabs, h_stabs)]
```

Comment:
Here we should probably have // instead of /, though the resulting floats seems to compare ok with Python ints.